### DISCUSSION TITLE
Color variables

### DISCUSSION THREAD

**Discussion by johnsicili** [webstudio-is/webstudio-community#48] [Wishlist]
Having color variables/saved colors speeds up development and makes it easy to globally update a color.

Would love to be able to:
- Save colors to be reused
- Give them a name (e.g., "Primary")
- Delete them
- Edit them

**Comment by kof** [comment]
Would you see color variables as a separate UI? I could imagine having a generic UI for variables like figma has that also works for colors

**Reply by ghost** [reply]
I believe he is referring to what Webflow has, which is the ability to make color swatches/variables through the color picker menu.

@johnsicili I agree this is a good idea. But it hasn't been prioritized because this functionality is already possible with tokens. Simply make a token that has no other styles except the color you want. Then you can apply that token to every element you wish. 

I know it's not as convenient as swatches/color variables but we will add those in time!

**Reply by kof** [reply]
I get it, but a color could be just as easily a variable, see figma variables

<img width="671" alt="Screenshot 2023-08-19 at 01 05 25" src="https://github.com/webstudio-is/webstudio/assets/52824/e7564127-f008-4cf2-be48-379ff51723e0">

**Reply by johnsicili** [reply]
Correct, my suggestion was the color swatches that Webflow has, or something similar where I can reference a defined color in the UI.

Yeah true @taylornowotny that's the best way to do it now. I think when this really comes into play is when I want to reuse a color for different properties.

So lets say I have my primary color that I want to easily reference when adding a border, a background, or text color. In this case I would need to go copy the color that I used on one of my tokens (e.g., background-primary) and then paste it into my new token (e.g,  border-primary). In this case it wouldn't make sense to create a token for each one, rather just have like a card token where I can easily select my primary color swatch from the UI.

**Reply by ghost** [reply]
Yes @kof the plan is for variables to have a range of data types, hex value for color being one of them. 

@johnsicili That would allow you to use the same color variable across multiple properties.

**Reply by kof** [reply]
I think color picker could just have a convenient access to color variables to pick a design system color directly from color picker

**Reply by ghost** [reply]
Yea that would be much better than swatches

**Comment by johnsicili** [comment]
Launched with CSS vars! https://www.youtube.com/watch?v=Th3XRwSltLs
